contracts/presets/oracles/ConstrainedOracle.sol

```diff
@@ -0,0 +1,146 @@
+pragma solidity >=0.8.0;
+
+import {BaseOracle} from "./BaseOracle.sol";
+import {InitializeOracleParams} from "../../commons/Types.sol";
+import {IKPIToken} from "../../interfaces/kpi-tokens/IKPIToken.sol";
+import {UNIT, INVALID_ANSWER} from "../../commons/Constants.sol";
+
+enum Constraint {
+    Between,
+    NotBetween,
+    // range is very similar to between, except the final completion
+    // percentage is determined based on the final absolute result
+    // checking where it landed in the specified range. As an example,
+    // if we have a range constraint with bounds 0 and 10 and the end
+    //  result is 5, the goal will be determined to have been 50% completed.
```

Comment:
Additional space here.

Reply:
Addressed in 085321c61ca8d07a7d23bc66030e6a08ff25b02e